packages/ui/src/lib/geolocation/Geocoder.svelte

```diff
@@ -0,0 +1,185 @@
+	import { MagnifyingGlass, XMark } from '@steeze-ui/heroicons';
+	import { Icon } from '@steeze-ui/svelte-icon';
+
+	import type { GeocoderAdapter } from './GeocoderAdapter';
+	import type {
+		GeolocationNamed, //
+		Geolocation,
+		GeolocationSearchError,
+		OnGeolocationSearchResult,
+		OnGeolocationSearchError
+	} from './types';
+
+	// adapter to source location suggestions from.
+	export let adapter: GeocoderAdapter;
+
+	// delay in ms after a key stroke to minimise redundant API requests.
+	export let delay = 250;
+
+	// onLocationSelected is called when a user clicks a location from the
+	// suggestions list.
+	export let onLocationSelected: undefined | OnGeolocationSearchResult;
+
+	// onSearchError is called when the adapter promise rejects a search request.
+	export let onSearchError: undefined | OnGeolocationSearchError;
+
+	// suggestions can be bound via 'bind:suggestions' to reactively receive
+	// changes to search results.
+	export let suggestions: GeolocationNamed[] = [];
+
+	// classes is a space separated list of additional classes applied to the
+	// root container.
+	export let classes = '';
+
+	// inputClasses is a space separated list of additional classes applied to
+	// the input element because laying input elements can be quite fiddly.
+	export let inputClasses = '';
+
+	// showClearButton forces the showing of the clear button at the end of the
+	// input element.
+	//
+	// It is set to false each time the user clicks the button so binding on this
+	// prop is useful.
+	export let showClearButton = false;
+
+	let container: HTMLElement;
+	let query = '';
+	let showSuggestionList = false;
+
+	const updateSuggestionsNow = async () => {
+		if (!query || query.length < 3) {
+			suggestions = [];
+			showSuggestionList = false;
+			return;
+		}
+
+		try {
+			suggestions = await adapter
+				.search(query)
+				.then((res) => res || [])
+				.catch((err: unknown) => {
+					console.error('[Location Search] Search suggestions could not be retrieved.');
+					onSearchError && onSearchError(err as GeolocationSearchError);
+					return [];
+				});
+
+			showSuggestionList = !!$$slots.default;
+		} catch (e: unknown) {
+			console.error(e);
+			showSuggestionList = false;
+		}
+	};
+
+	const [scheduleUpdate, cancelUpdate] = newDebouncer(updateSuggestionsNow, delay);
+
+	const closeSuggestionsList = () => {
+		cancelUpdate();
+		showSuggestionList = false;
+	};
+
+	const onSelect = (suggestion: Geolocation) => {
+		closeSuggestionsList();
+		onLocationSelected && onLocationSelected(suggestion);
+	};
+
+	const hideSuggestionList = (event: MouseEvent | KeyboardEvent) => {
+		const target = event.target as HTMLElement;
+		if (!target || !container || !isElementOrChildOf(target, container)) {
+			closeSuggestionsList();
+		}
+	};
+
+	const isElementOrChildOf = (element: HTMLElement, parent: HTMLElement) => {
+		const position = element.compareDocumentPosition(parent);
+		return !!(position & Node.DOCUMENT_POSITION_CONTAINS);
```

Comment:
```suggestion
		const positionBitmask = element.compareDocumentPosition(parent);
		return !!(positionBitmask & Node.DOCUMENT_POSITION_CONTAINS);
```

Renaming to `positionBitmask` makes it clear that this is a bitmask, and hence that bitwise AND is intended to query the status of a specific bit.